### ISSUE TITLE
RFE: out-of-source builds

### ISSUE THREAD

**Issue by avikivity** [fmtlib/fmt#424]
According to [this](https://cmake.org/Wiki/CMake_FAQ#Out-of-source_build_trees),

    cmake /path/to/fmt

should generate the makefile/build.ninja/whatever in the current directory; but instead it pollutes the source tree.

Can out of tree builds be supported?

**Comment by vitaut** [comment]
AFAICS out-of-source builds are already supported:

```
$ pwd
/tmp/foo
$ cmake /tmp/fmt
$ ls
CMakeCache.txt			cmake_install.cmake
CMakeFiles			doc
CPackConfig.cmake		fmt
CPackSourceConfig.cmake		fmt-config-version.cmake
CTestTestfile.cmake		fmt-config.cmake
Makefile			fmt-targets.cmake
...
```

What version do you use and what build files are generated in the source dir?

**Comment by avikivity** [comment]
I just tried HEAD.  Let me try again.

**Comment by avikivity** [comment]
Ah, it works after `git clean -fdx` in the source tree.  Sorry for the noise.

**Comment by vitaut** [comment]
No problem.
